editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -1106,6 +1107,53 @@ void CanvasItemEditor::_switch_theme_preview(int p_mode) {
 	EditorNode::get_singleton()->update_preview_themes(theme_preview);
 }
 
+void CanvasItemEditor::_prepare_translation_menu() {
+	const String current_preview_locale = EditorNode::get_singleton()->get_preview_locale();
+
+	translation_menu->clear();
+	translation_menu->reset_size();
+
+	translation_menu->add_radio_check_item(TTRC("None"));
+	translation_menu->set_item_metadata(-1, "");
+	if (current_preview_locale.is_empty()) {
+		translation_menu->set_item_checked(-1, true);
+	}
+
+	const Vector<String> locales = TranslationServer::get_singleton()->get_loaded_locales();
+	if (!locales.is_empty()) {
+		translation_menu->add_separator();
+	}
+	for (const String &locale : locales) {
+		const String name = TranslationServer::get_singleton()->get_locale_name(locale);
+		translation_menu->add_radio_check_item(name == locale ? name : name + " [" + locale + "]");
+		translation_menu->set_item_auto_translate_mode(-1, AUTO_TRANSLATE_MODE_DISABLED);
+		translation_menu->set_item_metadata(-1, locale);
+		if (locale == current_preview_locale) {
+			translation_menu->set_item_checked(-1, true);
+		}
+	}
+}
+
+void CanvasItemEditor::_switch_translation_preview(int p_index) {
+	for (int i = 0; i < translation_menu->get_item_count(); i++) {
+		translation_menu->set_item_checked(i, i == p_index);
+	}
+	EditorNode::get_singleton()->set_preview_locale(translation_menu->get_item_metadata(p_index));
+}
+
+void CanvasItemEditor::_translation_preview_changed() {
+	const String &locale = EditorNode::get_singleton()->get_preview_locale();
+
+	if (locale.is_empty()) {
+		translation_preview_button->hide();
+		return;
+	}
+
+	const String name = TranslationServer::get_singleton()->get_locale_name(locale);
+	translation_preview_button->set_text(vformat(TTR("Previewing: %s"), name == locale ? locale : name + " [" + locale + "]"));
```

Comment:
Nitpick, but this string won't update on editor's language change.
Maybe you could call `_translation_preview_changed()` on `NOTIFICATION_TRANSLATION_CHANGED`, if the button is visible?